cache.c

```diff
@@ -92,6 +92,7 @@ void* do_cache_alloc(cache_t *cache) {
                 free(ret);
                 object = NULL;
             }
+            cache->total++;
```

Comment:
yes, need to adjust total here! freecurr should be fine, since that ticks up only when there's enough space. The total/free counters are based on total and freecurr, so freetotal isn't touched here.